tests/simulationbase.py

```diff
@@ -39,8 +41,14 @@ def __init__(self, work_dir: Path) -> None:
     def init_provider(
             self,
             get_task_api_service: Callable[[Path], TaskApiService],
+            task_id
     ) -> None:
-        self.get_provider_task_api_service = get_task_api_service
+        self.mkdir_provider_task(task_id)
+        self._task_api_service = get_task_api_service(self.prov_task_work_dir)
+
+    def start_provider(self) -> None:
+        self.provider_client = ProviderAppClient(self._task_api_service)
```

Comment:
Any benefit in having this in a separate function instead of inside `init_provider` ?

Reply:
Yes, since the provider_client will shut down after each computation it needs to be `start`'ed more often then `init`'ed

Reply:
I see. Please put `provider` in the variable name tho.

Reply:
done